fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/SimpleObserver.java

```diff
@@ -184,7 +185,7 @@
     @PostConstruct
     public void buildListener() throws RepositoryException {
         LOGGER.debug("Constructing an observer for JCR events...");
-        session = repository.login();
+        session = getJcrSession(repository.login());
         session.getWorkspace().getObservationManager()
                 .addEventListener(this, EVENT_TYPES, "/", true, null, null, false);
         session.save();
```

Comment:
Not really. The short explanation is that the jcr-based event machinery is very closely tied to jcr Sessions, so the event listener needs direct access to a `jcr.Session`